### DISCUSSION TITLE
In example, why eslint and prettier are installed in root?

### DISCUSSION THREAD

**Discussion by Kazuhiro-Mimaki** [vercel/turborepo#5243] [Help] [answered]
In official `examples/basic`, eslint and prettier are installed in root. (https://github.com/vercel/turbo/blob/main/examples/basic/package.json)
But lint is configured in `/packages/eslint-config-custom`.
(https://github.com/vercel/turbo/tree/main/examples/basic/packages/eslint-config-custom)

Why eslint and prettier are not installed in `/packages/eslint-config-custom` but root? Is it impossible?

**Comment by thenbe** [comment, marked as answer]
Root package.json has [scripts](https://github.com/vercel/turbo/blob/2baad6fa7a3b3fda35130f9c53cedcb724fc931c/examples/basic/package.json#L6-L7) that rely on eslint and prettier, so they need to be installed in root.

> Why eslint and prettier are not installed in /packages/eslint-config-custom

Pnpm allows you to use a package in child workspaces [even if it's installed in the root workspace](https://pnpm.io/cli/run#details), but not the other way around.

This works fine. However, might be preferable to also add the eslint package a second time to the eslint-config-custom workspace. While this isn't necessary, it could help other tooling identify your dependency graph more accurately. pnpm dedupes them anyway, so there's no harm in installing it in multiple workspaces.
